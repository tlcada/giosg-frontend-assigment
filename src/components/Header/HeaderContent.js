import React from 'react';
import { withStyles } from '@material-ui/core/styles';

import { AppBar, Toolbar, Typography, IconButton } from '@material-ui/core';
import Gesture from '@material-ui/icons/Gesture';

const styles = {
    root: {
        flexGrow: 1,
    },
    grow: {
        flexGrow: 1,
    },
    menuButton: {
        marginLeft: -12,
        marginRight: 20,
    },
};


const HeaderContent = (props) => {
    const { classes } = props;

    return(
        <div className={classes.root}>
            <AppBar position="static">
                <Toolbar>
                    <IconButton className={classes.menuButton} color="inherit" aria-label="Menu">
                        <Gesture />
                    </IconButton>
                    <Typography variant="title" color="inherit" className={classes.grow}>
                        Giosg - Frontend Assignment
                    </Typography>
                </Toolbar>
            </AppBar>
        </div>
    )
};

export default withStyles(styles)(HeaderContent);
