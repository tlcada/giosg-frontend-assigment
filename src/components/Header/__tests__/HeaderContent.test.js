import React from 'react';
import { shallow } from 'enzyme';
import Header from '../HeaderContent';

describe('header', () => {
    it('renders without crashing', () => {
        const wrapper = shallow(<Header/>);
        expect(wrapper).toMatchSnapshot();
    });
});
