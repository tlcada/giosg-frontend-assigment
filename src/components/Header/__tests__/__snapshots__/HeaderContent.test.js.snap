// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(HeaderContent) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "classes": Object {
        "grow": "HeaderContent-grow-02",
        "menuButton": "HeaderContent-menuButton-03",
        "root": "HeaderContent-root-01",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Object {
          "grow": "HeaderContent-grow-02",
          "menuButton": "HeaderContent-menuButton-03",
          "root": "HeaderContent-root-01",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
